qrenderdoc/Windows/Dialogs/VirtualFileDialog.h

```diff
@@ -41,7 +42,8 @@ class VirtualFileDialog : public QDialog
   Q_OBJECT
 
 public:
-  explicit VirtualFileDialog(ICaptureContext &ctx, QString initialDirectory, QWidget *parent = 0);
+  explicit VirtualFileDialog(ICaptureContext &ctx, QString initialDirectory, QWidget *parent = 0,
+                             const rdcarray<rdcstr> &activityVector = rdcarray<rdcstr>());
```

Comment:
Hardcoding a platform-specific feature in here is not appropriate. The virtual file dialog already has handling to allow the data source to come from somewhere arbitrary on Android, adding specific hacks for a very niche use-case on an already niche platform adds potential for bugs and makes the code harder to read.

Reply:
The APK processing takes a little while, therefore it should be run as a background task. If we processed the APK in the `VirtualFileDialog`, the UI would be blocked by it.
The constructor of `VirtualFileDialog` should already have the data in order to set the variables. 
Can it work if we create a public method in the `VirtualFileDialog` for setting the data?

Reply:
As above this solution needs to work without slow APK processing. Even putting that aside, what you are saying doesn't make sense - background thread work can happen anywhere and my point is that the `VirtualFileDialog` should not need to know about ugly Android-specific details like activities, it should be provided data from elsewhere.

Moving the data from the constructor to a public method is a meaningless change as it is still entangled no matter what the function is.